File: app/src/main/java/com/example/notifications/MainActivity.java
package com.example.notifications;

import android.app.NotificationChannel;
import android.app.NotificationManager;
import android.app.PendingIntent;
import android.app.job.JobScheduler;
import android.app.job.JobService;
import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.os.Build;
import android.os.Bundle;
import android.os.Handler;
import android.support.v4.app.NotificationCompat;
import android.support.v4.app.NotificationManagerCompat;
import android.support.v4.app.RemoteInput;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {

    public static final String CHANNEL_ID = "5467811";
    Context context;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        context = this;
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button buttonCreateNotification = findViewById(R.id.button_create_notification);

        buttonCreateNotification.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                NotificationCompat.Builder builder = new NotificationCompat.Builder(view.getContext(), CHANNEL_ID)
                        .setSmallIcon(R.drawable.ic_launcher_foreground)
                        .setContentTitle("Simple Notification")
                        .setContentText("Come check out such and such things")
                        .setPriority(NotificationCompat.PRIORITY_DEFAULT);

                NotificationManagerCompat notificationManager = NotificationManagerCompat.from(view.getContext());
                // notificationId(123) is a unique int for each notification
                notificationManager.notify(123, builder.build());
            }
        });



    }
}

File: app/src/main/java/com/example/notifications/MainActivity.kt
package com.example.notifications

import android.app.NotificationManager
import android.app.PendingIntent
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.ImageView
import androidx.core.app.NotificationCompat
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    companion object{
        const val NOTIFICATION_ID = 87
        const val MESSAGE_ID = "46"
        const val OTHER_MESSAGE_ID = "84"
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        notification_btn.setOnClickListener {

            val intent = Intent(this, FullscreenActivity::class.java)
            intent.putExtra(MESSAGE_ID, "HELLO WORLD!")
            val pendingContentIntent = PendingIntent.getActivity(this, 0, intent, PendingIntent.FLAG_ONE_SHOT)

            intent.putExtra(OTHER_MESSAGE_ID, "WHY")
            val pendingActionIntent = PendingIntent.getActivity(this, 1, intent, PendingIntent.FLAG_ONE_SHOT)

            val notification = Notification(this, MainActivity::class.java)
            notification.notificationPriority = NotificationCompat.PRIORITY_LOW
            notification.setPendingContentIntent(pendingContentIntent)
            notification.addActionIntent(R.drawable.ic_launcher_background, "second intent", pendingActionIntent)

            notification.notify(NOTIFICATION_ID, getString(R.string.notification_title_main),
                getString(R.string.notification_content_main))
        }
    }
}
